content/docs/rfcs/37/README.md

```diff
@@ -0,0 +1,151 @@
+
+# Preliminaries
+
+We assume that two users, e.g. Alice and Bob, successfully completed a Noise handshake.
+
+Using [Noise terminology]((http://www.noiseprotocol.org/noise.html)), at the end of the handshake they will share:
+- two _Cipher States_ `CSOutbound` and `CSInbound`, to encrypt and decrypt outbound and inbound messages, respectively;
+- a handshake hash value `h`.
+
+As suggested in Noise specifications in regards to [Channel Binding](http://www.noiseprotocol.org/noise.html#channel-binding), 
+we can identify a Noise session with a `session-id` derived from the handshake hash value `h` shared on completion of a Noise handshake.
+
+More specifically, when Alice and Bob call [Split()](http://www.noiseprotocol.org/noise.html#the-symmetricstate-object) in order to derive the two final encryption and decryption Cipher States, 
+they further compute `session-id  = HKDF(h)` using the supported key derivation function `HKDF`.
+
+Such `session-id` will uniquely identify the Noise cryptographic session instantiated on completion of a Noise handshake,
+which would then consists of the tuple `(session-id, CSOutbound, CSInbound)`. 
+
+Once a Noise session is instantiated, 
+any further encrypted message between Alice and Bob within this session is exchanged on a `contentTopic` with name `/wakunoise/1/<ct-id>/proto`, 
+where `ct-id = Hash(Hash(session-id))`.
+
+
+# Session states
+
+A Noise session corresponding to a certain `session-id`:
+- is always **active** as long as it is not marked as **stale**. 
+For an active `session-id`, new messages are published on the content topic `/wakunoise/1/<ct-id>/proto`;  
+- is marked as **stale** if a [session termination message](https://rfc.vac.dev/spec/35/#session-termination-message) containing `Hash(session-id)` is published on the content topic `/wakunoise/1/<ct-id>/proto`. 
+Session information relative to stale sessions  MAY be deleted from users' device, unless required for later channel binding purposes.
+
+When a Noise session is marked as stale, it means that one party requested its termination while being online, 
+since publication of a hash pre-image for `ct-id` is required (i.e. `Hash(session-id)`).
+
+Currently, it is not possible to mark a Noise session as stale when `session-id` is lost or gets corrupted in users' devices.
+However, since `session-id` is shared between Alice and Bob, 
+one party MAY decide to mark a Noise session as stale if no message from the other end was received within a certain fixed time window. 
+
+The above mechanism allows a Noise session to be marked as stale either privately or publicly, 
+depending if `Hash(session-id)` is sent on `/wakunoise/1/<ct-id>/proto` to the other party in encrypted form or not, respectively. 
+
+When a Noise session is publicly marked as stale, 
+network peers MAY discard all [stored](https://rfc.vac.dev/spec/13/) messages addressed to the content topic `/wakunoise/1/<ct-id>/proto`. 
```

Comment:
Can you please clarify what network peers mean exactly?

Reply:
I think it would complicate things by asking store nodes to watch for the content topic of the waku messages. The content topic of a waku message should not have any specific meaning for the nodes running the store protocol otherwise they need to stay updated by the sematic of the content topic of all the applications. This only tangles the two protocols.

Reply:
network peers here refers mainly to store nodes (but I kept their definition generic, since the are not capable yet of that, as you described). We had internal discussion with @jm-clius, that in the near future store nodes may become able to watch the content topic of a Waku message. Hence the resources optimization given by staling sessions was designed having this possibility in mind, by considering also that this is a RAW initial proposal.

Reply:
> When a Noise session is publicly marked as stale, 
network peers MAY discard all [stored](https://rfc.vac.dev/spec/13/) messages addressed to the content topic `/wakunoise/1/<ct-id>/proto`.

I do not see this as a viable solution,  imagine a large number of applications running on top of waku and each coming up with a different semantic for the content topic namings, it is not reasonable to expect that store nodes to be aware of such external semantics and update their database/filter their incoming messages accordingly.  

> this is a RAW initial proposal.
I would suggest not including it in the RAW version

Reply:
Does this approach (already committed) help in this case? https://github.com/vacp2p/rfc/pull/504#discussion_r880283108